python/pyfury/_serialization.pyx

```diff
@@ -2049,7 +2050,13 @@ cdef class MapSerializer(Serializer):
         buffer.write_varint32(len(value))
         cdef ClassInfo key_classinfo
         cdef ClassInfo value_classinfo
-        for k, v in value.items():
+        cdef int64_t key_addr, value_addr
+        cdef Py_ssize_t pos = 0
+        while PyDict_Next(value, &pos, <PyObject **>&key_addr, <PyObject **>&value_addr) != 0:
+            k = int2obj(key_addr)
+            Py_INCREF(k)
```

Comment:
The references returned by `PyDict_Next` are borrowed; ideally, we wouldn't need any reference counting operations. However, since Cython automatically generates a `__Pyx_XDECREF` macro call for an object type variable at the end of the scope and during an assignment, we must manually call `Py_INCREF` to correct its reference count.

Reply:
I see, thanks.

Reply:
This auto generated operation is not that intuitive to me, maybe we should continue to consider refactoring cython code into other bindings like nanobind/pybind in #1887